fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/MetaDataPlanContext.java

```diff
@@ -108,33 +107,18 @@ public MetaDataPlanContext(@Nonnull RecordMetaData metaData, @Nonnull RecordStor
                 index -> !query.getAllowedIndexes().contains(index.getName()) :
                 index -> !query.getIndexQueryabilityFilter().isQueryable(index));
 
-        ImmutableSet.Builder<IndexEntrySource> builder = ImmutableSet.builder();
-        if (commonPrimaryKey != null) {
-            builder.add(IndexEntrySource.fromCommonPrimaryKey(recordTypes, commonPrimaryKey));
-        }
+        final ImmutableSet.Builder<MatchCandidate> matchCandidatesBuilder = ImmutableSet.builder();
         for (Index index : indexList) {
             indexes.put(index, index.getName());
-            builder.add(IndexEntrySource.fromIndex(metaData.recordTypesForIndex(index), index));
+            final Optional<MatchCandidate> candidateForIndexOptional =
+                    RelationalExpression.fromIndexDefinition(metaData, index, query.isSortReverse());
```

Comment:
I see how this works practically, but it seems odd to set the scan's reverse from the query's sort order. As opposed to trying various directions until one matches. Furthermore, if we ever had mixed order, the matching would need to be piecemeal.

Reply:
Discussed. We leave this as is right now. The proper way of doing this would be to create forward and reverse plans for every scan which doubles the search space effectively for no good reason as `isSortReversed` is a somewhat global parameter.